libs/core/src/components/sage-tab/sage-tab.tsx

```diff
@@ -0,0 +1,70 @@
+import { Component, Element, Host, h, Prop, Event, EventEmitter } from '@stencil/core';
+
+@Component({
+  tag: 'sage-tab',
+  styleUrl: 'sage-tab.scss',
+  shadow: false,
+})
+export class SageTab {
+  @Element() el: HTMLElement;
+
+  /**
+   * Sets the related tab name, this name must match a `sage-tabpanel`'s tab property
+  */
+  @Prop() tab!: string;
+
+  /**
+   * Keeps track of the parentComponent unique id, this property is passed by parent component
+  */
+  /** @internal */
+  @Prop({mutable: true}) parentComponent: string;
+
+  /**
+   * Keeps track of if the expected tab variant, this property is passed by parent component
+  */
+   /** @internal */
+  @Prop() variant: string;
```

Comment:
How does work in this scenario `@internal`? I ask because it looks like the `variant` is ultimately defined here even though it get passed through the the parent. Mainly trying to get more of an understanding since I would initially expect to see the corresponding test in the tab test. Being that the prop doesn't show on the `Docs` page, it them seems necessary to have the test in the tabs test file. Is this way preferred for items that contain subcomponents?

Reply:
I did some digging on the Ionic repo to see how they were handling props that weren't intended to be exposed to the people using components. They use `@internal` for that. I haven't been able to find good documentation on it but it doesn't seem to change any of the innerworkings of the components while hiding the props from the storybook args table automatically. 

In this usecase, that is exactly what we want. We want stencil to treat the `Prop()` decorator exactly the same but we don't want to expose Props to developers and confuse them. This allows us to only expose the props we expect developers to change and interact with. As you can't have just 1 tab with a different variant, with `Sage-Tab` and `Sage-Tabpanel` the variant is passed down from the parent intentionally. Exposing it to developers only adds confusion to our docs.

Reply:
I think this is what they're referencing: https://tsdoc.org/pages/tags/internal/

Reply:
Good find! Heres an example form the Ionic repo: https://github.com/ionic-team/ionic-framework/blob/545f0a9309bb453a78708bb382124ab285c041a2/core/src/components/router-outlet/route-outlet.tsx